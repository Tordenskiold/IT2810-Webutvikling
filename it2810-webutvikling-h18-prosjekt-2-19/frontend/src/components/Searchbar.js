import React from 'react';
import SearchBar from 'material-ui-search-bar';
import PropTypes from 'prop-types';
import { MuiThemeProvider } from 'material-ui';

class Searchbar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: null
    };
  }

//renders the searchbar in the website 
  render() {
    const { onSearch } = this.props;
    const { value } = this.state;
    return (
      <MuiThemeProvider>
        <SearchBar
        id="1"
          className="MainSearch"
          onChange={value => this.setState({ value })}
          onRequestSearch={() => onSearch(value)}
          style={{
            margin: '0 auto',
            marginTop: '40px',
            maxWidth: 800
          }}
        />
      </MuiThemeProvider>
    );
  }
}
Searchbar.propTypes = {
  onSearch: PropTypes.func.isRequired
};

export default Searchbar;
