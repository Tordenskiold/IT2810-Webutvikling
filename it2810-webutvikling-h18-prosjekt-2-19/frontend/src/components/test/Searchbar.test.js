import React from 'react';
import Searchbar from '../Searchbar';
import renderer from 'react-test-renderer';

//Jest test for Search.js and Store.js
//Creates a snapshot test which tests the rendering of the search bar using the elements from Store.js

it('renders without crashing', () => {
  const myMock = jest.fn();
  
  const component = renderer.create(
    <Searchbar onSearch = {myMock}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
