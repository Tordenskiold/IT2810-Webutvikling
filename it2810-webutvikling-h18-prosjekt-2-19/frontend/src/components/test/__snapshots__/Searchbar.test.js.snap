// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "height": 48,
      "justifyContent": "space-between",
      "margin": "0 auto",
      "marginTop": "40px",
      "maxWidth": 800,
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <div
    style={
      Object {
        "margin": "auto 16px",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "muiPrepared": true,
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className="MainSearch"
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 48,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 1,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Search
        </div>
        <input
          autoComplete="off"
          disabled={false}
          id="1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          open={false}
          spellCheck={false}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
  </div>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "marginRight": -48,
        "muiPrepared": true,
        "opacity": 0.54,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transform": "scale(1, 1)",
        "transition": "transform 200ms cubic-bezier(0.4, 0.0, 0.2, 1)",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "#9e9e9e",
            "height": 24,
            "muiPrepared": true,
            "opacity": 1,
            "transition": "opacity 200ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
  </button>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "muiPrepared": true,
        "opacity": 0.54,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transform": "scale(0, 0)",
        "transition": "transform 200ms cubic-bezier(0.4, 0.0, 0.2, 1)",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "#9e9e9e",
            "height": 24,
            "muiPrepared": true,
            "opacity": 0,
            "transition": "opacity 200ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </button>
</div>
`;
